### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by gandradep**
Develop class using test driven development.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Test Solver class and its methods' do
+    before(:each) do
+      @solve = Solver.new
+    end
```
Comment (t-yanick) [inline spec/solver_spec.rb:7]: - [ ] You have done a great job adding most of the test cases, but you missed the #factorial method test that handles a negative value.  It should throw an exception.  Kindly add that test case.

File: README.md [hunk 2]
```diff
@@ -99,123 +43,52 @@ After you're finished please remove all the comments and instructions!
...
-<!--
-Example command:
-
-```sh
-  rails server
-```
---->
+This repository includes the rspec folder that has the test files and the Solver class file.
 
 ### Run tests
```
Comment (t-yanick) [inline README.md:66]: - [ ] Kindly add a very clear procedure a user has to follow to run your project and the tests. This makes your ReadMe professional.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @gandradep, @sonyco-4u and @ndorrh , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (ndorrh) [reply spec/solver_spec.rb:7]: Thanks for review

File: README.md (hunk 2 above)
Comment (ndorrh) [reply README.md:66]: Thanks for the review

**Review by addod19 — APPROVED** [review]
### Hi TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### HIGHLIGHTS
- [ ] Nice job fixing the comments by the previous code reviewer :100: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
